### PR TITLE
feature(davide): add BSuite environment wrapper

### PR DESCRIPTION
**Pull request by davide97l**
## Description

This is a preliminary version. It contains a wrapper to make BSuite env compatible with DI-engine and relative tests. It also implements 'config' and 'entry' for 'memory_len' environment and 'config' for 'bandit_noise', both for DQN.

To know more about BSuite you can check the official GitHub repo: https://github.com/deepmind/bsuite

## Check List

- [x] merge the latest version source branch/repo, and resolve all the conflicts
- [ ] pass style check
- [ ] pass all the tests

### CODE REVIEW

File: dizoo/bsuite/envs/test_bsuite_env.py [hunk 1]
```diff
@@ -0,0 +1,57 @@
+from time import time
+import pytest
+import numpy as np
+from easydict import EasyDict
+from dizoo.bsuite.envs import BSuiteEnv
+
+
+@pytest.mark.unittest
```
Comment (PaParaZz1) [inline dizoo/bsuite/envs/test_bsuite_env.py:8]: we use `envtest` mark in test related to environment

Comment (davide97l) [reply dizoo/bsuite/envs/test_bsuite_env.py:8]: Received, I will modify accordingly. However I noticed that also other envs such as Atari use the `unittest ` mark, should it be changed?

File: dizoo/bsuite/entry/memory_len/memory_len_0_dqn_main.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+    collector_env = BaseEnvManager(env_fn=[memory_len_env for _ in range(collector_env_num)], cfg=cfg.env.manager)
+    evaluator_env = BaseEnvManager(env_fn=[memory_len_env for _ in range(evaluator_env_num)], cfg=cfg.env.manager)
+
+    # Set random seed for all package and instance
+    collector_env.seed(seed)
+    evaluator_env.seed(seed, dynamic_seed=False)
+    set_pkg_seed(seed, use_cuda=cfg.policy.cuda)
+
+    # Set up RL Policy
+    model = DQN(**cfg.policy.model)
```
Comment (zxzzz0) [inline dizoo/bsuite/entry/memory_len/memory_len_0_dqn_main.py:40]: Are we able to solve `memory_len/22` if not using any RNN model here?

Comment (davide97l) [reply dizoo/bsuite/entry/memory_len/memory_len_0_dqn_main.py:40]: No, I tried but DQN could only solve memory_len/0 and memory_len/22. I haven't tried other policies yet.

Comment (zxzzz0) [reply dizoo/bsuite/entry/memory_len/memory_len_0_dqn_main.py:40]: Okay.

File: dizoo/bsuite/envs/test_bsuite_env.py (hunk 1 above)
Comment (PaParaZz1) [reply dizoo/bsuite/envs/test_bsuite_env.py]: Yes, we will fix it in the following version.

**Review by PaParaZz1 — APPROVED** [review]
This PR can be merged now

**Pull request merged**